src/Centreon/Domain/ServiceConfiguration/ServiceConfigurationService.php

```diff
@@ -325,4 +334,31 @@ public function getHostConfigurationService(): HostConfigurationServiceInterface
     {
         return $this->hostConfigurationService;
     }
+
+    /**
+     * @param Host $host
+     * @throws ServiceConfigurationException
+     * @throws \Assert\AssertionFailedException
+     */
+    public function removeServices(Host $host): void
+    {
+        Assertion::notNull($host->getId(), 'Host::id');
+        try {
+            $this->debug('Remove services from a host', ['host_id' => $host->getId()]);
+            $this->serviceRepository->removeServicesOnHost($host->getId());
+        } catch (\Throwable $ex) {
+            $this->error(
+                sprintf(
+                    _('Error on removing services from the host #%d (Reason: %s)'),
+                    $host->getId(),
+                    $ex->getMessage()
+                )
+            );
+            throw new ServiceConfigurationException(
+                sprintf(_('Error on removing services from the host #%d'), $host->getId()),
+                0,
+                $ex
+            );
```

Comment:
Can you create a static method in `ServiceConfigurationException` to handle this Exception ?